### ISSUE TITLE
Spring Cloud Sleuth Tracing Configuration Issue on GKE

### ISSUE THREAD

**Issue by rajivml** [spring-cloud/spring-cloud-gcp#1569]
HI,

Am following the below article to setup distributed tracing for our Spring Boot Apps which are running on top of GKE.

The issue am facing is the traces which are getting generated are not visible via the Trace UI, am not able to debug the reason.

The example mentioned in the lab is working fine and I am able to see the traces, but when the same app is dockerized and when am running it on GKE am not able to see any traces via the Traces UI, I tried simulating 1000 requests but I couldn't see any of those.

https://codelabs.developers.google.com/codelabs/cloud-spring-cloud-gcp-trace/index.html?index=..%2F..index#0

Spring Boot Version: 2.1.1
Spring Cloud Dependencies Version: Greenwich.RELEASE
Trace.append enabled on cluster ? Yes
GKE Cluster Version ? 1.11.7-gke.4
Does Service Account has trace admin Role ? Yes
Docker Image Core ? gcr.io/distroless/java:8

Did You set GOOGLE_APPLICATION_CREDENTIALS environment variable in docker ?
Yes, this is how I have set the credentials and I logged into docker bash and verified that the environment variables are set properly,

kubectl create secret generic application-key --from-file=key.json=/home/rajiv/k8_rajiv/XXXX3.json

Kubernetes Template Excerpt On How GOOGLE_APPLICATION_CREDENTIALS environment variable is set:

      containers:
        - image: gcr.io/xxx-101/trace-one-server:default
          imagePullPolicy: Always
          name: trace-one-qa-deployment
          ports:
          - containerPort: 8080
          volumeMounts:
          - name: google-cloud-key
            mountPath: /var/secrets/google
          env:
          - name: GOOGLE_APPLICATION_CREDENTIALS
            value: /var/secrets/google/key.json

application.properties:

server.port=8081
spring.profiles.active=dev
spring.cloud.gcp.trace.enabled=true
spring.application.name=trace-one
#https://docs.spring.io/spring-cloud-gcp/docs/1.0.0.RELEASE/reference/htmlsingle/
spring.cloud.gcp.trace.project-id=sleuth-225414
spring.sleuth.sampler.probability=1.0
spring.sleuth.web.skipPattern=(^cleanup.*|.+favicon.*)  # Ignore some URL paths.
spring.rabbitmq.listener.direct.retry.enabled=true
spring.rabbitmq.listener.simple.retry.enabled=true
pring.cloud.gcp.trace.scheduled-delay-seconds=1

Pom Dependencies:

	<dependencies>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter</artifactId>
		</dependency>
		
		<dependency>
			<groupId>org.liquibase</groupId>
			<artifactId>liquibase-core</artifactId>
			<version>3.6.3</version>
		</dependency>
		
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-test</artifactId>
		</dependency>

		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-gcp-starter</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-gcp-starter-trace</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.boot</groupId>
			<artifactId>spring-boot-starter-web</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream-binder-rabbit</artifactId>
		</dependency>
		<dependency>
			<groupId>org.springframework.cloud</groupId>
			<artifactId>spring-cloud-stream-test-support</artifactId>
			<scope>test</scope>
		</dependency>
		<dependency>
			<groupId>org.projectlombok</groupId>
			<artifactId>lombok</artifactId>
			<optional>true</optional>
		</dependency>
	</dependencies>

Please let me know where else I could go wrong.

**Comment by dzou** [comment]
Hi @rajivml, thanks for reporting this issue. I'll take a deeper look at that article you linked and deploy it on GKE to see if I can identify the issue.

**Comment by dzou** [comment]
@rajivml - Could you share you Dockerfile for building app into an image?

**Comment by rajivml** [comment]
@dzou I have uploaded the entire sample demo project here https://drive.google.com/file/d/1OOqLoarHVHlZaL82UCgI1lnkSxv3knJ3/view?usp=sharing

**Comment by elefeint** [comment]
Possibly related to #1118 and #1096.

**Comment by dzou** [comment]
Hi @rajivml, I was able to get your sample to work by updating the version of spring-cloud-gcp to the latest release version: `1.1.1.RELEASE`

I've shared with you my private experimental repo with your code here: https://github.com/dzou/trace-service-one/blob/master/pom.xml

Unfortunately I am not sure why the Greenwich version did not work. I'll circle back with my team to see if we can identify a root cause - indeed this appears quite difficult to debug.

In any case, let me know if upgrading the version works for you.
